packages/core/src/external/carequality/ihe-gateway-v2/outbound/xca/process/dr-response.ts

```diff
@@ -52,7 +53,13 @@ async function parseDocumentReference({
   const strippedDocUniqueId = stripUrnPrefix(documentResponse.DocumentUniqueId);
   const metriportId = idMapping[strippedDocUniqueId];
   if (!metriportId) {
-    throw new Error("MetriportId not found for document");
+    throw new MetriportError("MetriportId not found for document");
+  }
+  if (!outboundRequest.patientId) {
+    throw new MetriportError("PatientId not found in outboundRequest");
+  }
+  if (!outboundRequest.cxId) {
+    throw new MetriportError("CxId not found in outboundRequest");
   }
```

Comment:
These will result in runtime error. We could turn it into compile time errors if we keep the outbound request type requiring cxId and patientId and make the inbound request type having those as optional.